File: 3876_Construct_Uniform_Parity_Array_II/sol.java
import java.util.*;

class Solution {
    /**
     * Time:    O(N)
     * Space:   O(N)
     * Strategy: Minimum tracking with condtion filtering/ Two-pass baseline approach.
     */
    public boolean solve(int[] nums1) {
        int mini = Integer.MAX_VALUE;
        for (int i : nums1) {
            mini = Math.min(i, mini);
        }
        if (mini %2 ==0) {
            for (int i : nums1) {
                if (i %2 !=0) {
                    return false;
                }
            }
        }
        return true;
    }
    public static void main(String[] args) {
        Solution sol = new Solution();
        int[][] inputs = {
            {1, 4, 7},
            {2, 3},
            {4, 6},
        };
        boolean[] expected = {true, false, true};
        System.out.println("Running Tests...");
        for (int i = 0; i < inputs.length; i++) {
            boolean result = sol.solve(inputs[i]);
            String status = (result == expected[i]) ? "✅ PASS" : "❌ FAIL";
            System.out.printf("%s | Test #%d: Input: %s | Expected: %B | Result: %B%n",
                status, i+1, Arrays.toString(inputs[i]), expected[i], result );
        }
    }
}

File: 3876_Construct_Uniform_Parity_Array_II/sol.js
/**
 * @param {number[]} nums1
 * @return {boolean}
 * Time:    O(N)
 * Space:   O(N)
 * Strategy: Minimum tracking with condtion filtering/ Two-pass baseline approach.
 */
var func = function(nums1) {
    let min = Number.MAX_SAFE_INTEGER;
    for (const x of nums1) {
        min = Math.min(min, x);
    }
    if(min %2 ===0){
        for (const x of nums1) {
            if(x %2 !==0){
                return false;
            }
        }
    }
    return true;
};
/**
 * Local testing
 */
function test() {
    const testCases = [
        { args: [1,4,7], expected: true },
        { args: [2,3], expected: false },
        { args: [4,6], expected: true }
    ];
    testCases.forEach(({ args, expected }, index) => {
        const result = func(args);
        const status = (result === expected) ? "✅ PASS" : "❌ FAIL";
        console.log(
            `${status} | Test #${index + 1}: ` +
            `Input: ${JSON.stringify(args)} | ` +
            `Expected: ${expected} | ` +
            `Result: ${result}`
        );
    });
}
test();